directives/tabs.py

```diff
@@ -0,0 +1,129 @@
+# This implementation is based on the sphinxcontrib-contentui repository:
+# https://github.com/ulrobix/sphinxcontrib-contentui/
+# -*- coding: utf-8 -*-
+"""
+Tabs
+====
+
+Directive for creating tabs.
+
+.. content-tabs::
+
+  .. tab-container:: tab1
+    :title: Tab title one
+
+    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus hendrerit 
+    auctor quam at maximus. Phasellus ornare suscipit tortor et aliquet. Aliquam 
+    erat volutpat. Aliquam at orci vel nibh tincidunt lacinia. Cras gravida, 
+    mauris eget vulputate ullamcorper, turpis est commodo velit, consequat 
+    pulvinar augue mi malesuada metus. Maecenas ac diam et augue placerat 
+    faucibus. Nullam ut iaculis nisi.
+
+  .. tab-container:: tab2
+    :title: Tab title two
+
+    .. code-block:: rst
+    
+      Insert Your Code here.
+"""
+import os
+from docutils.parsers.rst import Directive, directives
+from docutils import nodes
+from docutils.statemachine import StringList
+from sphinx.util.fileutil import copy_asset
+from pkg_resources import resource_filename
+from sphinx.util import logging
+
+import aplus_nodes
+
+CSS_FILE = 'css/aplus_tabs.css'
+JS_FILE = 'js/aplus_tabs.js'
+assets_path = '_static'
+
+logger = logging.getLogger(__name__)
+
+
+class ContentTabsDirective(Directive):
+    has_content = True
+    optional_arguments = 1
+
+    def run(self):
+        self.assert_has_content()
+        text = '\n'.join(self.content)
+        node = nodes.container(text)
+        node['classes'].append('content-tabs')
+
+        if self.arguments and self.arguments[0]:
+            node['classes'].append(self.arguments[0])
+
+        self.add_name(node)
+        self.state.nested_parse(self.content, self.content_offset, node)
+        return [node]
+
+
+class ContentTabsContainerDirective(Directive):
+    has_content = True
+    option_spec = {'title': directives.unchanged}
+    required_arguments = 1
+
+    def run(self):
+        self.assert_has_content()
+        text = '\n'.join(self.content)
+        node = nodes.container(text)
+        node['ids'].append('tab-%s' % self.arguments[0])
+        node['classes'].append('tab-content')
+
+        par = nodes.paragraph(text=self.options["title"])
+        par['classes'].append('tab-title')
+        node += par
+
+        self.add_name(node)
+        self.state.nested_parse(self.content, self.content_offset, node)
+
+        return [node]
+
+
+# Files added after triggering the 'builder-inited' event, in order to read the
+# settings from the conf.py file and prevent from reading the settings that
+# were added in the setup function
+def add_assets(app):
+    app.config.aplus_rst_tools_tabs_css and app.add_stylesheet(CSS_FILE)
+    app.config.aplus_rst_tools_tabs_js and app.add_javascript(JS_FILE)
+
+
+def copy_asset_files(app, exception):
+    if exception is None:
```

Comment:
I already commented on this function in the annotated PR #82 and some comments apply here too.